src/util/memory.h

```diff
@@ -9,11 +9,11 @@
 #include <memory>
 #include <utility>
 
-//! Substitute for C++14 std::make_unique.
+//! No longer needed substitute for C++14 std::make_unique. Needed when we were using C++11.
 template <typename T, typename... Args>
 std::unique_ptr<T> MakeUnique(Args&&... args)
```

Comment:
Could add `deprecated` attribute and remove them in the following few PRs.
```suggestion
[[deprecated]] std::unique_ptr<T> MakeUnique(Args&&... args)
```

Reply:
It would throw warning [C4996](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4996) and we treat it as an error in MSVC. Maybe you could add 4996 in MSVC's warning ignore list.

Reply:
@ken2812221 My suggestion is that we simply remove `MakeUnique` in a `scripted-diff` follow-up PR as soon as this PR is merge as discussed in https://github.com/bitcoin/bitcoin/pull/15262#discussion_r251110176. Makes sense?

Reply:
> @ken2812221 My suggestion is that we simply remove MakeUnique in a scripted-diff follow-up PR as soon as this PR is merged as discussed in #15262 (comment). Makes sense?

To be honest, if it's a scripted diff we could just as well do it in one go (in a sep commit) here. No need to drag this out, it's not controversial in any way once we've decided to go to C++14.

Reply:
@laanwj That's how I did it initially in this PR but @MarcoFalke suggested it should be done in a follow-up PR -- see comment https://github.com/bitcoin/bitcoin/pull/15262#discussion_r251110176 :-)

Reply:
The rationale behind this was that any pull requests that are expected to be open for "longer" should minimize any (silent) merge conflicts.

Reply:
I agree with that @MarcoFalke. Especially since it looks like this PR will be open for quite a while?

Reply:
> The rationale behind this was that any pull requests that are expected to be open for "longer" should minimize any (silent) merge conflicts.

Mostly I'd agree. But OTOH if the merge conflict is in the scripted-diff part then rebasing should be trivial.

I don't like the prospect of follow-up PR after follow-up PR. Better IMO to do it at once. But okay.

> I agree with that @MarcoFalke. Especially since it looks like this PR will be open for quite a while?

Let's merge it after the 0.18 split.